modules/programs/fish.nix

```diff
@@ -0,0 +1,124 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.fish;
+
+  abbrsStr = concatStringsSep "\n" (
+    mapAttrsToList (k: v: "abbr --add ${k} '${v}'") cfg.shellAbbrs
+  );
+
+  envVarsStr = config.lib.shell.exportAll cfg.sessionVariables;
+
+  aliasesStr = concatStringsSep "\n" (
+    mapAttrsToList (k: v: "alias ${k}='${v}'") cfg.shellAliases
+  );
+
+in
+
+{
+  options = {
+    programs.fish = {
+      enable = mkEnableOption "fish friendly interactive shell";
+
+      shellAliases = mkOption {
+        default = {};
+        description = ''
+          Set of aliases for fish shell. See <option>environment.shellAliases</option>
+          for an option format description.
+        '';
+        type = types.attrs;
+      };
+      
+      shellAbbrs = mkOption {
+        default = {};
+        description = ''
+          Set of abbrs for fish shell. See <option>environment.shellAliases</option>
```

Comment:
Best to avoid abbreviations in the description 😉

So, I'd suggest to start the sentence with "Set of abbreviations".